massa-module-cache/gas-table.md

```diff
@@ -0,0 +1,61 @@
+# Overview
+
+This document keeps track of where and how gas costs are handled for SC executions.
+
+It contains:
+* the gas table
+* the list of functions that effectively pay the costs
+
+# Gas table
+
+| TYPE                     | PAYMENT                             |
+| ------------------------ | ----------------------------------- |
+| **CallSC (OP)**          |                                     |
+| compilation              | NOT PAID                            |
+| VM instantiation         | PAID IN MAX_GAS (UNIFIED CONSTANT)  |
+| module instantiation     | PAID IN MAX_GAS                     |
+| execution                | PAID IN MAX_GAS                     |
+| **ExecuteSC (OP)**       |                                     |
+| compilation              | PAID IN MAX_GAS                     |
```

Comment:
wouldn't it be more consistent if it is paid in the [base op cost](https://github.com/massalabs/massa/pull/4510) instead ?

Reply:
Same as https://github.com/massalabs/massa/pull/4505#discussion_r1381893641, that means having different types of behaviours for every node operation using tmp modules when we only have 1 atm

Reply:
Again no strong opinion, can change it if you want

Reply:
Also this is related to your comment above: https://github.com/massalabs/massa/pull/4505#issuecomment-1792619101

This is also a good point, I'm okay with both will let you have the final say

Reply:
I tend to prioritize user-facing consistency over code factorization